### ISSUE TITLE
Unable to set bgp_measurements = 0 in agent to server test

### ISSUE THREAD

**Issue by cgascoig** [thousandeyes/terraform-provider-thousandeyes#59]
I'm trying to use this provider to manage some agent to server tests for internal addresses but kept getting this error: "There were some errors in your request, please correct them before trying again. Can only use private BGP Monitors for a local address."

I tried disabling both bgpMeasurements and usePublicBgp:
```
  bgp_measurements = 0
  use_public_bgp = 0
```
But it didn't make any difference. 

I rebuilt the provider so I could add some logging in go-thousandeyes and it seems that the API request is sent without the usePublicBgp and bgpMeasurements attributes:
```
{
  "alertsEnabled": 1,
  "alertRules": [
    {
      "ruleId": XYZ
    },
    {
      "ruleId": XYZ
    }
  ],
  "enabled": 1,
  "testName": "XYZ",
  "agents": [
    {
      "agentId": XYZ
    }
  ],
  "bandwidthMeasurements": 1,
  "interval": 600,
  "mtuMeasurements": 1,
  "networkMeasurements": 1,
  "numPathTraces": 3,
  "pathTraceMode": "classic",
  "probeMode": "AUTO",
  "protocol": "ICMP",
  "server": "XYZ"
}
```

Digging a little deeper looks like this is because the UsePublicBgp and BgpMeasurements fields in go-thousandeyes have "omitempty" [set](https://github.com/william20111/go-thousandeyes/blob/130ff0ecdcb28f8212e6925086939aa67f603858/agent_server.go#L32) - so when they are explicitly set to 0 in Terraform, they are omitted from the API request and the TE API defaults to 1. 

I think this exposes a broader issue in this provider - in my specific case, removing "omitempty" from these specific fields would solve my problem. However, this issue would seem to exist in all the integer/boolean attributes. Removing "omitempty" from all of them would effectively invert the default from the TE API documentation (i.e. omitting bgp_measurements from the TF config would cause bgpMeasurements to be explicitly disabled where the API specifies that it is enabled by default). 

I suppose this issue could be addressed a couple of ways:
* Keep the structs in go-thousandeyes the same and remove "omitempty" where appropriate, then in the TF schema ensure the default is set correctly (not sure how feasible this is as the schema appears to be generated from the structs)
* Change the structs in go-thousandeyes to *int instead of int so that "unset", 0 and 1 states can be correctly differentiated

**Comment by robertcsapo** [comment]
This is related to how Terraform handles non-zero values.
https://github.com/william20111/terraform-provider-thousandeyes/blob/2efa6a2f628a044552876dbc2591faf90989fb24/thousandeyes/util.go#L114

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.5.0/helper/schema#ResourceData.GetOk

This also causes a block to use Cloud Agents with Terraform.
As they need ```bandwidth_measurements = 0```

> "set to 1 to measure bandwidth. Only applies to Enterprise Agents assigned to the test, and requires that networkMeasurements is set."

**Comment by raul-te** [comment]
This issue has been fixed in #64.
